react-components/src/architecture/base/commands/BaseCommand.ts

```diff
@@ -0,0 +1,100 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+/* eslint-disable @typescript-eslint/class-literal-property-style */
+
+import { clear, remove } from '../utilities/extensions/arrayExtensions';
+
+type UpdateDelegate = () => void;
+
+export type Tooltip = {
+  key: string;
+  fallback: string;
+};
+
+export abstract class BaseCommand {
+  // ==================================================
+  // INSTANCE FIELDS
+  // ==================================================
+
+  private readonly _listeners: UpdateDelegate[] = [];
+
+  // ==================================================
+  // VIRTUAL METHODS (To be override)
+  // =================================================
+
+  public get name(): string {
+    return this.tooltip.fallback;
+  }
+
+  public get shortCutKey(): string | undefined {
```

Comment:
I'm skeptical to let the tools define themselves whether they have a shortcut key. I think the application should be in absolute control of this, and not be riddled with predefined short-cuts from react-components as a library.

Or am I misunderstanding how this works?

Reply:
Not really. If you have a better solution on this, come with it. But my message from Per Øyvind was to make the apps equal. If the apps has different shortCut, we will fail.